### PR TITLE
Kalunge/tf 32 GitHub User Authentication

### PR DESCRIPTION
**Pull request by Kalunge**
Add GitHub user auth using firebase

### CODE REVIEW

File: src/hooks/useLoginWithGithub.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const [isPending, setIsPending] = useState(false);
+  const navigate = useNavigate()
+
+  const loginWithGithub = useCallback(async () => {
+    setError("");
+    setIsPending(true);
+    const provider = new GithubAuthProvider();
+    try {
+      const res = await signInWithPopup(auth, provider);
+        console.log(res);
```
Comment (otsembo) [inline src/hooks/useLoginWithGithub.ts:17]: Remove the console log.

File: src/hooks/useLoginWithGithub.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        if(!res.user){
+            throw new Error("No user found");
+        }
+        const user = res.user
+        console.log(user)
```
Comment (otsembo) [inline src/hooks/useLoginWithGithub.ts:22]: Remove the console log.

File: src/hooks/useLoginWithGithub.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        setIsPending(false);
+        navigate("/");
+    } catch (err:  any) {
+        console.log(err)
```
Comment (otsembo) [inline src/hooks/useLoginWithGithub.ts:26]: Unnecessary log.

**Review by otsembo — APPROVED** [review]
Hello @Kalunge ,
Good Job on the PR. Kindly check on the minor fixes.

**Review by mutuajoseph — APPROVED** [review]

**Pull request merged**